source/MaterialXCore/Document.cpp

```diff
@@ -611,7 +611,7 @@ void Document::upgradeVersion()
         InterfaceElementPtr interfaceElem = std::static_pointer_cast<InterfaceElement>(nodeDef);
         if (interfaceElem && interfaceElem->hasType())
         {
-            string type = interfaceElem->getType();
+            string type = interfaceElem->getAttribute(TypedElement::TYPE_ATTRIBUTE);
```

Comment:
I don't understand. This seems to do what getType() does?

Reply:
That's what I forgot about as well, remember we overrode getType for NodeDefs.

Reply:
Ok. This is to check for the deprecated "type" attribute.